protocols/v2/noise-sv2/src/initiator.rs

```diff
@@ -274,6 +379,8 @@ impl Initiator {
     }
 }
 impl Drop for Initiator {
+    /// Ensures that sensitive data is securely erased when the [`Initiator`] instance is dropped,
```

Comment:
as a rule of thumb, I agree that we shouldn't be writing docs for trait implementations, as we usually get them from the trait definition

I believe this specific case is an exception, because we are providing valuable insight into how `drop` can be used as a safety mechanism

Reply:
Note that it is already mentioned in the section at L59

Reply:
Makes sense @jbesraa , removing